packages/core/_src/stm/TMap/operations/takeFirst.ts

```diff
@@ -0,0 +1,61 @@
+import { concreteTArray } from "@effect/core/stm/TArray/operations/_internal/InternalTArray"
+import { concreteTMap } from "@effect/core/stm/TMap/operations/_internal/InternalTMap"
+
+/**
+ * Takes the first matching value, or retries until there is one.
+ *
+ * @tsplus fluent ets/TMap takeFirst
+ */
+export function takeFirst_<K, V, A>(
+  self: TMap<K, V>,
+  pf: (kv: Tuple<[K, V]>) => Option<A>
+): USTM<A> {
+  concreteTMap(self)
+  return STM.Effect<never, Option<A>>((journal) => {
+    let result = Option.emptyOf<A>()
```

Comment:
Can we use `Option.none` here and assert the type? Just to avoid the extra allocation.